client/src/components/ProtectedRoute/ProtectedRoute.tsx

```diff
@@ -0,0 +1,22 @@
+import { Route, Redirect, RouteProps } from 'react-router-dom';
+import { useAuth } from '../../context/useAuthContext';
+
+export default function ProtectedRoute({ ...props }: RouteProps): JSX.Element {
+  const { loggedInUser } = useAuth();
+  const redirectPath = '/login';
+
```

Comment:
Additionally, we can add a loading spinner for the split second it takes to fetch the logged in user's information when we refresh the page